framework/db/ActiveQueryTrait.php

```diff
@@ -111,45 +111,20 @@ public function with()
      */
     protected function createModels($rows)
     {
-        $models = [];
         if ($this->asArray) {
-            if ($this->indexBy === null) {
-                return $rows;
-            }
-            foreach ($rows as $row) {
-                if (is_string($this->indexBy)) {
-                    $key = $row[$this->indexBy];
-                } else {
-                    $key = call_user_func($this->indexBy, $row);
-                }
-                $models[$key] = $row;
-            }
+            return $rows;
         } else {
+            $models = [];
```

Comment:
I know there's no need but I find it more readable, early return statements tend to make your code less readable; is it in violation of the coding  standard?